### PR TITLE
Restructure docs organization and naming and add Javadoc

### PR DESCRIPTION
**Pull request by jlin27**
- Rename “Other Languages” → “Language Bindings”
- Move the Community section to the bottom
- Move "Language Bindings" above "Python API"
- Add Javadoc url in index.rst
- Delete no longer needed java rst files
- Remove intersphinx extension

### CODE REVIEW

**Comment by zou3519** [comment]
Is this change in master? We should get it into master first and then merge it into 1.4.0.

(it doesn't look like it is in master:
image
)

**Comment by jlin27** [comment]
Here is the PR that was created for master that contains all the same changes except adding the Javadoc url in index.rst: https://github.com/pytorch/pytorch/pull/30757

Could you help us review and merge this one first?

**Comment by kostmo** [comment]
## :pill: CircleCI build failures summary and remediations
As of commit 7e21db20:


**None of the build failures appear to be your fault**.

* **1/1** recognized as flaky :snowflake:
    * Re-run these jobs?


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=7e21db20ac4c305e27705830b2191aa83ecf7a01).

### :snowflake: 1 failure recognized as flaky
The following build failures have been detected as flaky and **may not be your fault**:
#### [See CircleCI build](https://circleci.com/gh/pytorch/pytorch/4091280) pytorch_macos_10_13_py3_test (1/1)
**Step:** "Test" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=78789699) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=78789699)) :snowflake:
<details>
<summary>
<code>Dec 30 16:24:32 RuntimeError: test_quantized_nn_mods failed! Received signal: SIGSEGV</code>
</summary>

```
Dec 30 16:24:30 Generating XML reports... 
Dec 30 16:24:30 Running test_quantized_nn_mods ... [2019-12-30 16:24:30.477967] 
Dec 30 16:24:31  
Dec 30 16:24:31 Running tests... 
Dec 30 16:24:31 ---------------------------------------------------------------------- 
Dec 30 16:24:32 s...Traceback (most recent call last): 
Dec 30 16:24:32   File "test/run_test.py", line 455, in <module> 
Dec 30 16:24:32     main() 
Dec 30 16:24:32   File "test/run_test.py", line 448, in main 
Dec 30 16:24:32     raise RuntimeError(message) 
Dec 30 16:24:32 RuntimeError: test_quantized_nn_mods failed! Received signal: SIGSEGV 
Dec 30 16:24:32 + cleanup 
Dec 30 16:24:32 + retcode=1 
Dec 30 16:24:32 + set +x 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

**Review by zou3519 — APPROVED** [review]

File: docs/source/conf.py [hunk 1]
```diff
@@ -46,7 +46,6 @@
     'sphinx.ext.autodoc',
     'sphinx.ext.autosummary',
     'sphinx.ext.doctest',
-    'sphinx.ext.intersphinx',
```
Comment (zou3519) [inline docs/source/conf.py]: We should also remove javaspinx from docs/requirements.txt . Are the java docs removal changes in master as well?

**Review by zou3519 — CHANGES_REQUESTED** [review]

File: docs/source/conf.py (hunk 1 above)
Comment (jlin27) [reply docs/source/conf.py]: I deleted the fork this PR was made from so I'll need to make a new PR with javasphinx change

Comment (zou3519) [reply docs/source/conf.py]: go for it

**Review by zou3519 — APPROVED** [review]

**Pull request merged**